database/migrations/2024_06_18_123547_add_nested_set_columns_to_digital_publication_articles.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use App\Models\DigitalPublicationArticle;
+
+return new class () extends Migration {
+    public function up()
+    {
+        Schema::table('digital_publication_articles', function (Blueprint $table) {
+            $table->nestedSet();
+        });
+
+        // Get unique digital_publication_ids
+        $digitalPublicationIds = DigitalPublicationArticle::select('digital_publication_id')
+            ->distinct()
+            ->pluck('digital_publication_id');
+
+        foreach ($digitalPublicationIds as $digitalPublicationId) {
+            // Get articles for this digital_publication_id, ordered by position
+            $articles = DigitalPublicationArticle::where('digital_publication_id', $digitalPublicationId)
+                ->orderBy('position')
+                ->get();
+
+            // Initialize the left value
+            $lft = 1;
+
+            foreach ($articles as $article) {
+                // Set the left and right values for this article
+                $article->_lft = $lft;
+                $article->_rgt = $lft + 1;
```

Comment:
I think you can assign each of these values the value `$lft++`. That should assign the variable to the current `$lft` value, then increment it once the statement is resolved. Then you can remove the `$lft += 2;` statement below.

Reply:
@nikhiltri This doesn't retain the initial iteration of $lft = 1 which breaks the tree. I've made updates to the migration file which uses the ++ but not in this way.